### PR TITLE
Fix for example with Retina displays

### PR DESCRIPTION
**Pull request by spite**
Account for device pixel ratio (either multiply the mouse coordinates, or use the pixel size of the canvas instead of the CSS size of the element)

### CODE REVIEW

**Comment by spite** [comment]
There's a ton of old commits in this PR, I don't know why GH decided to add them :-?

File: examples/webgl_geometry_terrain_raycast.html [hunk 1]
```diff
@@ -252,8 +252,8 @@
 
 			function onMouseMove( event ) {
 
-				mouse.x = ( event.clientX / renderer.domElement.width ) * 2 - 1;
-				mouse.y = - ( event.clientY / renderer.domElement.height ) * 2 + 1;
+				mouse.x = ( ( event.clientX * window.devicePixelRatio ) / renderer.domElement.width ) * 2 - 1;
+				mouse.y = - ( ( event.clientY * window.devicePixelRatio ) / renderer.domElement.height ) * 2 + 1;
```
Comment (WestLangley) [inline examples/webgl_geometry_terrain_raycast.html]: Try this, instead.
```javascript
mouse.x = ( event.clientX / renderer.domElement.clientWidth ) * 2 - 1;
mouse.y = - ( event.clientY / renderer.domElement.clientHeight ) * 2 + 1;
```

Comment (mrdoob) [inline examples/webgl_geometry_terrain_raycast.html]: @spite ping?

Comment (spite) [inline examples/webgl_geometry_terrain_raycast.html]: Whichever way is best. I can't do a PR ATM. Can you change it on your end?

Comment (leitzler) [inline examples/webgl_geometry_terrain_raycast.html]: Created pull request #7280, hope you don't mind @spite

Comment (leitzler) [inline examples/webgl_geometry_terrain_raycast.html]: I also verified that the suggestion by @WestLangley works.

**Comment by mrdoob** [comment]
#7280

**Pull request closed without merging**